### ISSUE TITLE
Can't locate object method "license_class" via package "Dist::Zilla::Stash::Rights"

### ISSUE THREAD

**Issue by hakonhagland** [rjbs/Dist-Zilla#645]
I am trying to build [this](https://github.com/jrockway/anyevent-subprocess) module with `dzil build`, but it throws an error:

```
$ dzil build
[DZ] beginning to build AnyEvent-Subprocess
Can't locate object method "license_class" via package "Dist::Zilla::Stash::Rights" at /home/hakon/perlbrew/perls/perl-5.30.0/lib/site_perl/5.30.0/Dist/Zilla.pm line 407, <DATA> line 8.
```

By inspecting [line 407](https://github.com/rjbs/Dist-Zilla/blob/master/lib/Dist/Zilla.pm#L407) in `Dist/Zilla.pm` I can see that the `$stash` variable indeed does not have a method `license_class()` but it has a hash field `license_class`, so if I change the line from

```
$stash && return $stash->license_class;
```

to 

```
$stash && return $stash->{license_class};
```

it works. Any idea why this happens?

**Comment by karenetheridge** [comment]
I can't reproduce that using a recent install of Dist::Zilla and all plugins in use.  The only change I had to make was `s/Test/TestRequires/` in dist.ini on the `[Prereqs]` line.

**Comment by karenetheridge** [comment]
> Can't locate object method "license_class" via package "Dist::Zilla::Stash::Rights" at /home/hakon/perlbrew/perls/perl-5.30.0/lib/site_perl/5.30.0/Dist/Zilla.pm line 407, <DATA> line 8.

lib/Dist/Zilla/Stash/Rights.pm does indeed have a `license_class` method, which it gets via a Moose attribute.

**Comment by hakonhagland** [comment]
@karenetheridge `/Test/TestRequires/` Yes that was [the first thing I did](https://github.com/jrockway/anyevent-subprocess/pull/11). After I did that fix I got the error *"Can't locate object method "license_class"..."* Since, it works for you it must be something with my setup? I installed `Dist::Zilla` with `cpanm` which gave me version 6.012. I can try with the latest version of `Dist::Zilla` from GitHub..

**Comment by hakonhagland** [comment]
Ok, I did a test with the latest version of `Dist::Zilla`:
```
git clone git@github.com:rjbs/Dist-Zilla.git
cd Dist-Zilla
dzil authordeps --missing | cpanm
dzil install
```
Now I changed directory to the `anyevent-subprocess` fork Git repository and did:

```
$ dzil build
[DZ] beginning to build AnyEvent-Subprocess
Can't locate object method "license_class" via package "Dist::Zilla::Stash::Rights" at /home/hakon/perlbrew/perls/perl-5.30.0/lib/site_perl/5.30.0/Dist/Zilla.pm line 407, <DATA> line 8.
```

So the same problem as before. If I put a breakpoint a line 407 and run:
```
$ perl -d /home/hakon/perlbrew/perls/perl-5.30.0/bin/dzil build
Loading DB routines from perl5db.pl version 1.55
Editor support available.

Enter h or 'h h' for help, or 'man perldebug' for more help.

dzil::(/home/hakon/perlbrew/perls/perl-5.30.0/bin/dzil:8):
8:	$main::VERSION = $Dist::Zilla::App::VERSION;
  DB<1> c
[DZ] beginning to build AnyEvent-Subprocess
Dist::Zilla::CODE(0x5650aa07a218)(/home/hakon/perlbrew/perls/perl-5.30.0/lib/site_perl/5.30.0/Dist/Zilla.pm:408):
408:	    $stash && return $stash->license_class;
  DB<1> x $stash
0  Dist::Zilla::Stash::Rights=HASH(0x5650ad2e9250)
   'copyright_holder' => 'Jonathan Rockway'
   'copyright_year' => 2019
   'license_class' => 'Perl_5'
  DB<2> p $stash->license_class
Can't locate object method "license_class" via package "Dist::Zilla::Stash::Rights" at (eval 1531)[/home/hakon/perlbrew/perls/perl-5.30.0/lib/5.30.0/perl5db.pl:738] line 2, <DATA> line 8.
```
Very strange, since when I inspect the file [`lib/Dist/Zilla/Stash/Rights.pm`](https://github.com/rjbs/Dist-Zilla/blob/master/lib/Dist/Zilla/Stash/Rights.pm) I can see that the class indeed has a method with that name...

**Comment by hakonhagland** [comment]
The problem seems to be that the module `Dist::Zilla::Stash::Rights` has not yet been loaded when the thread of execution reaches line 407. If I insert a `use Dist::Zilla::Stash::Rights;` it works.. Any idea why the module has not been loaded?

**Comment by hakonhagland** [comment]
I think I found another clue, if I add a config file for `dzil` in my home directory, as `~/.dzil/config.ini`:
```
[%Rights]
license_class    = Perl_5
copyright_holder = Håkon Hægland
```
it also works. So it appears that the `%Rights` section in the `config.ini` causes `dzil` to load the module `Dist::Zilla::Stash::Rights` ? But is this something I have missed? Is it documented that the user should set up a `config.ini` before running `dzil build`? It seems like a bug to me, since the copy right settings should not be loaded from my config, but according to the module authors `dist.ini` right?

**Comment by karenetheridge** [comment]
Yes -- if I remove my ~/.dzil/config.ini, I now get the same error as you.

However, I can still build other distributions -- I did some tracing and found that in those distributions, Dist::Zilla::Stash::Rights isn't being loaded at all.  That's because the distribution sets 'license' at the top of dist.ini (which all distributions ought to do) -- when that field is missing, dzil falls back to your local config to see what your preferred licence is.  Of course, that's not the same as what the *author*'s preferred licence is. :)

tldr: AnyEvent-Subprocess's dist.ini really needs to declare a licence.  IMHO Dist::Zilla itself should not fall back to the local config file here, but instead die with a helpful error message.

**Comment by hakonhagland** [comment]
Added a [pull request](https://github.com/jrockway/anyevent-subprocess/pull/12) to `anyevent-subprocess`.
